# Project: Investigate and Analysis a CSV Dataset <br>"IT Salary Survey for EU region (2020)"

### Data Link : https://www.kaggle.com/datasets/parulpandey/2020-it-salary-survey-for-eu-region

## Table of Contents
<ul>
<li><a href="#dd">Data Discovery</a></li>
<li><a href="#dw">Data Wrangling And Cleaing</a></li>
<li><a href="#df">Data Final Look</a></li>
<li><a href="#eda">Exploratory Data Analysis</a></li>
<li><a href="#con">Conclusions</a></li>
<li><a href="#st">The Story</a></li>

</ul>



# About The Data

An anonymous salary survey has been conducted annually since 2015 among European IT specialists with a stronger focus on Germany.<br>
it Contains 1238 respondents volunteered to participate in the survey.<br>


# Our Goal From The DataSet: 
* Analysis The IT Market Of Opportunity and How To Find a Job as IT Specialist in Europe 2020 
* How To Prepare Yourself as Student and have a High Chace to Get A Job As IT Specialist

## EDA Questions:
<ol>
    
    
    
<li>How Big Are The Companies In Europe ?

    
<li>What Are The Most Popular Technologies Per Each Position ?

<li>How Many Years Of Experience Do You Need To Get Each Work Level In Germany?

<li>Which City has The Highest Chance To Get An Opportunity?

<li>Which Domain Pays more ?

<li>Which Age Has The Highest Chance To Get an IT Opportunity ?

<li>How safe Are The Opportunites in Europe ?                 
    
<li>What is The most Common Work Level ?

<li>Which Work Level Get More Vacation ?

<li>Which Language Should You Use There ?
    
<li>What is The Most Popluar Job In Europe 2020?

    
<li>How Covid-19 Affects on The Jobs in 2020 ?  

</ol>

# Columns Description 

**Timestamp:** Row's Recorded Time <br>

**Age:** Person's Age <br>

**Gender:** Person's Gender <br>

**City:** Person's City <br>

**Position:** Person's job Positoon <br>

**Total years of experience:** Person's Total Experience <br>

**Years of experience in Germany:** Person's Germany Experience <br>

**Seniority level:** Person's Work Level <br>

**Your main technology / programming language:** Person's Main Work Technology <br>

**Other technologies/programming languages you use often:** Person's Other Technology Can Be Used <br>

**Yearly brutto salary (without bonus and stocks) in EUR:** Person's Main Salary <br>

**Yearly bonus + stocks in EUR:** Person's Main Salary With The bonus <br>

**Annual brutto salary (without bonus and stocks) one year ago. Only answer if staying in the same country:** Person's Annual brutto salary <br>

**Annual bonus+stocks one year ago. Only answer if staying in same country:** Person's bonus of the last year <br>

**Number of vacation days:** vacation days Per Year <br>

**Employment status:** Job State <br>

**Сontract duration:** Contract State <br>

**Main language at work:** Main Used Language <br>

**Company size:** Company size <br>

**Company type:** Company Domain <br>

**Have you lost your job due to the coronavirus outbreak?:** Corona Affect On The Person's Job <br>

**Have you been forced to have a shorter working week (Kurzarbeit)? If yes, how many hours per week:** If The Person forced to have a shorter working week <br>

**Have you received additional monetary support from your employer due to Work From Home? If yes, how much in 2020 in EUR:** If The Worker Forced The Employee To Work More due to Work From Home<br>


## Import And Load The CSV File 

In [1]:
import pandas as pd
import numpy as np
import plotly.express as px
from plotly.offline import init_notebook_mode, iplot
init_notebook_mode(connected=True)
import matplotlib.pyplot as plt
import seaborn as sn

In [2]:
# Load The CSV File
df = pd.read_csv("IT Salary Survey EU  2020.csv")
df.head()

Timestamp   Age Gender    City           Position   \
0  24/11/2020 11:14:15  26.0   Male  Munich   Software Engineer   
1  24/11/2020 11:14:16  26.0   Male  Berlin   Backend Developer   
2  24/11/2020 11:14:21  29.0   Male  Berlin   Software Engineer   
3  24/11/2020 11:15:24  28.0   Male  Berlin  Frontend Developer   
4  24/11/2020 11:15:46  37.0   Male  Berlin   Backend Developer   

  Total years of experience Years of experience in Germany Seniority level  \
0                         5                              3          Senior   
1                         7                              4          Senior   
2                        12                              6            Lead   
3                         4                              1          Junior   
4                        17                              6          Senior   

  Your main technology / programming language  \
0                                  TypeScript   
1                                        Ruby   
2                     Javascript / Typescript   
3                                  Javascript   
4                                     C# .NET   

  Other technologies/programming languages you use often  ...  \
0                    Kotlin, Javascript / Typescript      ...   
1                                                NaN      ...   
2                    Javascript / Typescript, Docker      ...   
3                                                NaN      ...   
4                             .NET, SQL, AWS, Docker      ...   

   Annual bonus+stocks one year ago. Only answer if staying in same country  \
0                                              10000                          
1                                               5000                          
2                                             100000                          
3                                                NaN                          
4                                                NaN                          

  Number of vacation days           Employment status   Сontract duration  \
0                      30          Full-time employee  Unlimited contract   
1                      28          Full-time employee  Unlimited contract   
2                      30  Self-employed (freelancer)  Temporary contract   
3                      24          Full-time employee  Unlimited contract   
4                      29          Full-time employee  Unlimited contract   

  Main language at work Company size Company type  \
0               English       51-100      Product   
1               English     101-1000      Product   
2               English     101-1000      Product   
3               English       51-100      Startup   
4               English     101-1000      Product   

  Have you lost your job due to the coronavirus outbreak?  \
0                                                 No        
1                                                 No        
2                                                Yes        
3                                                 No        
4                                                 No        

  Have you been forced to have a shorter working week (Kurzarbeit)? If yes, how many hours per week  \
0                                                NaN                                                  
1                                                NaN                                                  
2                                                NaN                                                  
3                                                NaN                                                  
4                                                NaN                                                  

  Have you received additional monetary support from your employer due to Work From Home? If yes, how much in 2020 in EUR  
0                                                NaN                                    

<a id='dd'></a>

# Data Discovery

<img src="https://media.giphy.com/media/l2JhORT5IFnj6ioko/giphy.gif
"/>

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1253 entries, 0 to 1252
Data columns (total 23 columns):
 #   Column                                                                                                                   Non-Null Count  Dtype  
---  ------                                                                                                                   --------------  -----  
 0   Timestamp                                                                                                                1253 non-null   object 
 1   Age                                                                                                                      1226 non-null   float64
 2   Gender                                                                                                                   1243 non-null   object 
 3   City                                                                                                                     1253 non-null   o

<a id='dw'></a>
# Data Wrangling And Cleaing

<img src="https://media2.giphy.com/media/eMbKluW9vLGtFKI5JR/giphy.gif?cid=790b76115a267806e51ce17aa44eab93764caf217ee82620&rid=giphy.gif&ct=g
"/>

**Some Columns Need To Get Removed, Because No Use For Them:** <br>
1- "Have you been forced to have a shorter working week (Kurzarbeit)? If yes, how many hours per week"<br>
2- "Have you received additional monetary support from your employer due to Work From Home? If yes, how much in 2020 in EUR"<br>
3- "Annual bonus+stocks one year ago. Only answer if staying in same country"<br>
4- "Yearly bonus + stocks in EUR"<br>
5- "Annual brutto salary (without bonus and stocks) one year ago. Only answer if staying in the same country"<br>
6- 'Timestamp'

In [4]:
df.drop(
    columns=[
        "Have you been forced to have a shorter working week (Kurzarbeit)? If yes, how many hours per week",
        "Have you received additional monetary support from your employer due to Work From Home? If yes, how much in 2020 in EUR",
        "Annual bonus+stocks one year ago. Only answer if staying in same country",
        "Yearly bonus + stocks in EUR",
        "Annual brutto salary (without bonus and stocks) one year ago. Only answer if staying in the same country",
        'Timestamp'],
    inplace = True 
)

**Rename Some Columns For Easier Use**

In [5]:
df.rename(columns = {
    'Total years of experience':'experience',
    'Years of experience in Germany':'germany_experience',
    'Seniority level':'work_level',
    'Your main technology / programming language':'main_tech',
    'Other technologies/programming languages you use often':'other_tech',
    'Number of vacation days':'vacation_days',
    'Yearly brutto salary (without bonus and stocks) in EUR':'yearly_salary',
    'Employment status':'emp_state',
    'Сontract duration':'contract_duration',
    'Main language at work':'main_language',
    'Company size':'company_size',
    'Company type':'company_type',
    'Have you lost your job due to the coronavirus outbreak?':'covid_lost',
    'Age':'age',
    'Gender':'gender',
    'City':'city',
    'Position ':'position'
    
}, inplace = True)

In [6]:
df.head()

age gender    city            position experience germany_experience  \
0  26.0   Male  Munich   Software Engineer          5                  3   
1  26.0   Male  Berlin   Backend Developer          7                  4   
2  29.0   Male  Berlin   Software Engineer         12                  6   
3  28.0   Male  Berlin  Frontend Developer          4                  1   
4  37.0   Male  Berlin   Backend Developer         17                  6   

  work_level                main_tech                       other_tech  \
0     Senior               TypeScript  Kotlin, Javascript / Typescript   
1     Senior                     Ruby                              NaN   
2       Lead  Javascript / Typescript  Javascript / Typescript, Docker   
3     Junior               Javascript                              NaN   
4     Senior                  C# .NET           .NET, SQL, AWS, Docker   

   yearly_salary vacation_days                   emp_state  \
0        80000.0            30          Full-time employee   
1        80000.0            28          Full-time employee   
2       120000.0            30  Self-employed (freelancer)   
3        54000.0            24          Full-time employee   
4        62000.0            29          Full-time employee   

    contract_duration main_language company_size company_type covid_lost  
0  Unlimited contract       English       51-100      Product         No  
1  Unlimited contract       English     101-1000      Product         No  
2  Temporary contract       English     101-1000      Product        Yes  
3  Unlimited contract       English       51-100      Startup         No  
4  Unlimited contract       English     101-1000      Product         No

## Check Structure Error And Null Values

In [7]:
for i in df.columns:
    print(
        i,': Null Count is',df[i].isnull().sum(),
          '\n',df[i].unique(),
          '\n','*'*50
    )

age : Null Count is 27 
 [26. 29. 28. 37. 32. 24. 35. nan 34. 31. 41. 27. 25. 59. 36. 38. 40. 39.
 33. 30. 49. 48. 44. 66. 45. 43. 42. 46. 47. 56. 53. 65. 22. 23. 50. 51.
 21. 20. 54. 69. 52.] 
 **************************************************
gender : Null Count is 10 
 ['Male' 'Female' nan 'Diverse'] 
 **************************************************
city : Null Count is 0 
 ['Munich' 'Berlin' 'Hamburg' 'Wolfsburg' 'Stuttgart' 'Schleswig-Holstein'
 'London' 'Konstanz area' 'Frankfurt' 'Cologne' 'Kempten' 'Münster'
 'Erlangen' 'Vienna' 'Moldova' 'Rosenheim' 'Mannheim ' 'Boeblingen'
 'Düsseldorf' 'Ingolstadt' 'Nürnberg' 'Ansbach' 'Leipzig' 'Mannheim'
 'Tuttlingen' 'Bonn' 'Moscow' 'Koblenz' 'Warsaw' 'Heidelberg' 'Karlsruhe'
 'Köln' 'Aachen' 'Karlsruhe ' 'Samara' 'Riga, Latvia' 'Dusseldorf'
 'Zurich' 'Helsinki' 'Würzburg' 'Kiev' 'Den Haag' 'Amsterdam' 'Cracovia'
 'Tallinn' 'Prague' 'Utrecht' 'Stockholm' 'Braunschweig ' 'Dresden' 'Kyiv'
 'Stuttgart ' 'Malta' 'Lübeck' 'Nuremberg ' 'Bod

## Work On Structure Error

In [8]:
# covid_lost Column
dics ={
"i didn't but will be looking for new one because of covid":'No',
'kurzarbeitzeit for 1.5 months':'No',
'Have been a freelancer at the beginning of year':'No',
'No, but there was a salary cut at 10% for 3 months and then at 5% for further 3 months':'No',
'Lost the job but for different reason':'No',
'Leads and project inquiries have slowed down' :'No',
'Laid off for a bit':'No',
'yes but found a new one with better pay / perks':'Yes' 
}

for i in dics:
    df['covid_lost'] = df['covid_lost'].apply(lambda x:dics.get(i) if (x==i) else x )

In [9]:
# main_language Column
dics = {
'English and German':'English',
'50/50':'English',
'Русский':'Russian',
'both':'English',
'Russian, English':'English'
}

for i in dics:
    df['main_language'] = df['main_language'].apply(lambda x:dics.get(i) if (x==i) else x )

In [10]:
# contract_duration Column
dics = {
'0':'Temporary contract'
}

for i in dics:
    df['contract_duration'] = df['contract_duration'].apply(lambda x:dics.get(i) if (x==i) else x )

In [11]:
# emp_state Column
dics = {
'Full-time employee':'full-time',
'Self-employed (freelancer)':'freelancer',
'Company Director':'full-time',
'Founder':'full-time',
'Part-time employee':'part-time',
'working student':'part-time',
'Working Student':'part-time',
"Full-time position, part-time position, & self-employed (freelancing, tutoring)":'full-time',
"full-time, but 32 hours per week (it was my request, I'm a student)":'part-time',
'Werkstudent':'part-time'
}

for i in dics:
    df['emp_state'] = df['emp_state'].apply(lambda x:dics.get(i) if (x==i) else x )

In [12]:
# Experience Column
dics = {
    '1,5':'1.50',
    '1 (as QA Engineer) / 11 in total':'11',
    '2,5':'2.5',
    '15, thereof 8 as CTO':'15',
    '6 (not as a data scientist, but as a lab scientist)':'6',
    '383':'38',
    'less than year':'0'
}

for i in dics:
    df['experience'] = df['experience'].apply(lambda x:dics.get(i) if (x==i) else x )

In [13]:
# Germany_experience  Column
dics = {'1,5':'1.5',
'4,5':'4.5',
'1,7':'1.7',
'3,5':'3.5',
'< 1':'0',
'4 (in Switzerland), 0 (in Germany)':'0',
'4 month':'0',
'2,5':'2.5',
'0,5':'0.5',
'3 months':'0',
'3 (in Poland)':'0',
'<1':'0',
'⁰':'0',
'0,3':'0',
'-':'0',
'6 (not as a data scientist, but as a lab scientist)':'0',
'0.2':'0',
'less than year':'0'
}


for i in dics:
    df['germany_experience'] = df['germany_experience'].apply(lambda x:dics.get(i) if (x==i) else x )

In [14]:
# work_level Column
dics = {
    'Intern':'intern',
    'Student':'student',
    "no idea, there are no ranges in the firm ":'Senior'
}


for i in dics:
    df['work_level'] = df['work_level'].apply(lambda x:dics.get(i) if (x==i) else x )

In [15]:
# vacation_days  Column
dics = {
    '(no idea)':'0',
    '30 in contract (but theoretically unlimited)':'30',
    'Unlimited ':'100',
    '~25':'25',
    '23+':'25',
    'Unlimited':'100',
    '24 labour days':'24',
    'unlimited ':100,
    'unlimited':100
}


for i in dics:
    df['vacation_days'] = df['vacation_days'].apply(lambda x:dics.get(i) if (x==i) else x )

## Deal With Null Values

Fill Age Column Null Values Based On Average Experience of Each Age

In [16]:
df['age'] = df['age'].fillna(df.groupby('experience')['age'].transform('mean'))

# Drop Rows With Null Experiance
df.dropna(subset=['age'], how='all', inplace=True)

<b>Change age datatype to use it on filling experience null rows

In [ ]:
df['age']=df['age'].astype('int64')
df['experience']=df['experience'].astype('float')
df['germany_experience']=df['germany_experience'].astype('float')

Drop Rows Of Null Gender Column

In [ ]:
df.dropna(subset=['gender'], how='all', inplace=True)

Drop Rows Of Null Position Column

In [ ]:
df.dropna(subset=['position'], how='all', inplace=True)

Fill Experience Column Null Values Based On Average Age of Each Experience

In [ ]:
df['experience'] = df['experience'].fillna(df.groupby('age')['experience'].transform('mean'))

# Drop Rows With Null Age
df.dropna(subset=['experience'], how='all', inplace=True)

check outliers on germany_experience to fill nulls

In [ ]:
print('Mean',df.germany_experience.mean(),'and Median',df.germany_experience.median())
px.box(df.germany_experience).show()

Use Median To Fill Nulls

<b> Drop Nulls on Those Columns Because No Reference To Get The Data From It

In [ ]:
df.germany_experience.fillna(df.germany_experience.median,inplace=True)

In [ ]:
df.dropna(subset=['other_tech'], how='all', inplace=True)

In [ ]:
df.dropna(subset=['main_tech'], how='all', inplace=True)

In [ ]:
df.dropna(subset=['vacation_days'], how='all', inplace=True)

In [ ]:
df.dropna(subset=['contract_duration'], how='all', inplace=True)

In [ ]:
df.dropna(subset=['company_type'], how='all', inplace=True)

In [ ]:
df.dropna(subset=['contract_duration'], how='all', inplace=True)

In [ ]:
df.dropna(subset=['work_level'], how='all', inplace=True)

In [ ]:
df.dropna(subset=['covid_lost'], how='all', inplace=True)

In [ ]:
df.dropna(subset=['emp_state'], how='all', inplace=True)

In [ ]:
df.dropna(subset=['main_language'], how='all', inplace=True)

In [ ]:
df.dropna(subset=['company_size'], how='all', inplace=True)

In [ ]:
# convert vacation number column to float
df['vacation_days']=df['vacation_days'].astype('float')

<a id='df'></a>
# Data Final Look

In [ ]:
df.head()

In [ ]:
df.info()

In [ ]:
df.describe()

<a id=eda></a>
# Exploratory Data Analysis

<img src="https://media2.giphy.com/media/y31rRE5h3wyPXey8vx/giphy.gif?cid=ecf05e47tyfles51nr6kpluybtefs5d01lwgaqf53oiprkyt&rid=giphy.gif&ct=g">

## 1.How Big Are The Companies In Europe ?

In [ ]:
%%HTML
<div class='tableauPlaceholder' id='viz1663511626954' style='position: relative'><noscript><a href='#'><img alt='The Most Common Companies Size  ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it13&#47;compsizes&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='it13&#47;compsizes' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it13&#47;compsizes&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1663511626954');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>    

## 2.What Are The Most Popular Technologies Per Each Position ?

In [ ]:
%%HTML
<div class='tableauPlaceholder' id='viz1663511715715' style='position: relative'><noscript><a href='#'><img alt='Most Used Tech For Positions  ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it12&#47;usedtech&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='it12&#47;usedtech' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it12&#47;usedtech&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1663511715715');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>
    

## 3.How Many Years Of Experience Do You Need To Get Each Work Level In Germany?

In [ ]:
%%HTML
<div class='tableauPlaceholder' id='viz1663511805579' style='position: relative'><noscript><a href='#'><img alt='Average Germany Experience In Each Work Level ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it10&#47;gerexp&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='it10&#47;gerexp' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it10&#47;gerexp&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1663511805579');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

## 4.Which City has The Highest Chance To Get An Opportunity?

In [ ]:
%%HTML
<div class='tableauPlaceholder' id='viz1663511926075' style='position: relative'><noscript><a href='#'><img alt='High Chance City To Find a IT Opportunity ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it9&#47;city&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='it9&#47;city' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it9&#47;city&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1663511926075');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

## 5.Which Domain Pays more ?

In [ ]:
%%HTML
<div class='tableauPlaceholder' id='viz1663512057683' style='position: relative'><noscript><a href='#'><img alt='Top Company Type At The Salary ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it8&#47;topsalary&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='it8&#47;topsalary' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it8&#47;topsalary&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1663512057683');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

## 6.Which Age Has The Highest Chance To Get an IT Opportunity ?

In [ ]:
%%HTML
<div class='tableauPlaceholder' id='viz1663512318170' style='position: relative'><noscript><a href='#'><img alt='Range Of Ages IT Agence in 2020 ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it7&#47;rangeofages&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='it7&#47;rangeofages' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it7&#47;rangeofages&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1663512318170');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

## 7.How safe Are The Opportunites in Europe ?

In [ ]:
%%HTML
<div class='tableauPlaceholder' id='viz1663512392103' style='position: relative'><noscript><a href='#'><img alt='Contract Type ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it6&#47;contract&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='it6&#47;contract' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it6&#47;contract&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1663512392103');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

## 8.What is The most Common Work Level ?

In [ ]:
%%HTML
<div class='tableauPlaceholder' id='viz1663512837113' style='position: relative'><noscript><a href='#'><img alt='Percentage Of Each Work Level ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it5&#47;worklevel&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='it5&#47;worklevel' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it5&#47;worklevel&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1663512837113');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>
    

## 9.Which Work Level Get More Vacation ?

In [ ]:
%%HTML
<div class='tableauPlaceholder' id='viz1663512882713' style='position: relative'><noscript><a href='#'><img alt='Average Vacation Days Per Position ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it4&#47;vacation&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='it4&#47;vacation' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it4&#47;vacation&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1663512882713');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

## 10.Which Language Should You Use There ?

In [ ]:
%%HTML
<div class='tableauPlaceholder' id='viz1663512927092' style='position: relative'><noscript><a href='#'><img alt='Which Language Is The Most Popular ?  ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it3_16634168893850&#47;language&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='it3_16634168893850&#47;language' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it3_16634168893850&#47;language&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1663512927092');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

## 11.What is The Most Popluar Job In Europe 2020?

In [ ]:
%%HTML 
<div class='tableauPlaceholder' id='viz1663514281043' style='position: relative'><noscript><a href='#'><img alt='Most Common Jobs at IT in 2020 ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it2_16634168514120&#47;commonjobs&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='it2_16634168514120&#47;commonjobs' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it2_16634168514120&#47;commonjobs&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1663514281043');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

## 12.How Covid-18 Affects on The Jobs in 2020 ?  

In [ ]:
%%HTML
<div class='tableauPlaceholder' id='viz1663513011500' style='position: relative'><noscript><a href='#'><img alt='How Covid Affects The Jobs ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it1&#47;covid&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='it1&#47;covid' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;it&#47;it1&#47;covid&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1663513011500');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

 <a id='con'></a>
 # Conclusion

<br>**How Big Are The Companies In Europe ?** <br>
* 36% of them over 1000 Employee
* 32% of them 100-1000 Employee
* 14% of them 10-50 Employee
* 12% of them 50-110 Employee
* 5% of them up to Employee

<br>**What Are The Most Popular Technologies Per Each Position ?** <br>
* Backend Developer
    - Java
    - PHP
    - Python
    
<br>

* Data Analyst, Data Engineer, Data Scientist, and ML Engineer
    - Pthon
    
<br>

* DevOps
    - Python
    - Java

<br>

* Software Engineer
    - Java
    - Python
    - PHP

<br>

* Mobile Depeloper
    - Java

<br>

* Project Manager
    - Python


<br>**How Many Years Of Experience Do You Need To Get Each Work Level In Germany?** <br>
* 13 Years For a C-Level
* 7 Years For a VP
* 5.5 Years For a Head
* 5 Years For a lead
* 4.5 Years For Senior
* 4 Years For CTO
* 1.5 Years For a Junior

<br>**Which City has The Highest Chance To Get An Opportunity?** <br>
* Berlin
* Munich
* Hamburg

<br>**Which Domain Pays more ?** <br>
* Concern
* Cloud
* Pharma
* E-Commerce
* Corporation

<br>**Which Age Has The Highest Chance To Get an IT Opportunity ?** <br>
* the Range Between 25 To 38 Is The Most Luckly To Get The Job

<br>**How safe Are The Opportunites in Europe ?** <br>
* 94% Of The Comntracte Are Unlimited Contract, So It's Really Safe 

<br>**What is The most Common Work Level ?** <br>
* Senior, Then Middle, Then Lead

<br>**Which Work Level Get More Vacation ?** <br>
* Consaltant
* Team Lead
* QA Engineer
* Solution Architect
* Manager

<br>**Which Language Should You Use There ?** <br>
* English
* German

<br>**What is The Most Popluar Job In Europe 2020?** <br>
* Software Engineer
* Backend-Developer
* Data Scientist

<br>**How Covid-19 Affects on The Jobs in 2020 ?** <br>
* On 2020, Covid Has Not The Effect on The IT Jobs

 <a id='st'></a>
 # The Story

<img src='https://media.giphy.com/media/XrwenGNux1GpYu25Qp/giphy.gif'>

#### If You Want To Get a Job As IT Specialist Here is Some tips that wiLL Help.
1- You Should Be Between 25 and 38 Is The Most Luckily To Get The Job.<br>
2- You Should Know Java Or Python Because It's The Most Used Programming Langauge<br>
3- You Should Search Berlin, Munich, or Hamburg Because They Have The Highest Chance Of Find A Job There<br>
4- You Should Search on Concern, Cloud, Pharma, E-Commerce, or Corporation Domains Because They Pay The Most.<br>
5- You Must BE  Perfect With English, Because It's The Most Used Language Then Germany<br>
6- You Should Learn To be Software Engineer, Backend-Developer, or Data Scientist For Higher Job opportunities.<br>
<br>
Don't Worry About Losing Your Job Die To Covid-19, because it doesn't Affect On It.<br>
Don't Worry About The Job Security, Because 94% Of The Contract Are Unlimited Contract<br>

<img style="display:inline; " src="https://media4.giphy.com/media/lQ6KHKqQaPrx7CBhPD/giphy.gif?cid=790b7611f9ec670f52232fd75b6b5273e13f44c70a07c032&rid=giphy.gif&ct=g" width="300" height="400">

<img style="display:inline" src="https://media2.giphy.com/media/BYoRqTmcgzHcL9TCy1/giphy.gif?cid=790b76117545874ff11cb1e1099fbc05dcbd5567b77a1735&rid=giphy.gif&ct=g"/>

<img style="display:inline; " src="https://media4.giphy.com/media/lQ6KHKqQaPrx7CBhPD/giphy.gif?cid=790b7611f9ec670f52232fd75b6b5273e13f44c70a07c032&rid=giphy.gif&ct=g" width="300" height="400">
